Code/Framework/AzCore/Tests/Memory/PoolAllocatorTests.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzCore/Memory/PoolAllocator.h>
+#include <AzCore/std/containers/vector.h>
+
+constexpr size_t s_testPoolPageSize = 1024;
+constexpr size_t s_testPoolMinAllocationSize = 64;
+constexpr size_t s_testPoolMaxAllocationSize = 256;
+
+// Define a custom pool allocator
+class TestPoolAllocator final : public AZ::Internal::PoolAllocatorHelper<AZ::PoolSchema>
```

Comment:
This class name needs to be different to avoid issue with Unity files combining both `VectorAndArray.cpp` and `PoolAllocatorTests.cpp` into a single unit